api/s2n.h

```diff
@@ -344,6 +344,10 @@ extern ssize_t s2n_client_hello_get_extension_length(struct s2n_client_hello *ch
 S2N_API
 extern ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tls_extension_type extension_type, uint8_t *out, uint32_t max_length);
 S2N_API
+extern ssize_t s2n_client_hello_get_raw_extension_length(struct s2n_client_hello *ch, uint16_t extension_type);
+S2N_API
+extern ssize_t s2n_client_hello_get_raw_extension_by_id(struct s2n_client_hello *ch, uint16_t extension_type, uint8_t *out, uint32_t max_length);
```

Comment:
I'm not sure these are the right APIs to solve the problem. Should we give customers access to extensions that s2n-tls hasn't parsed / validated / looked at in any way? It might be safer to just implement the s2n_client_hello_has_extension method the issue calls out, or even make that more specific as s2n_client_hello_has_unknown_extension. Or we could name these extensions get_unparsed/unvalidated_extension to make them very scary.

Reply:
After some discussion, two conclusions: 1) Only provide a "has" method for now, not a "get" method 2) Make the new "has" method work for both supported and unsupported extensions. We can use [this method](https://github.com/aws/s2n-tls/blob/main/tls/extensions/s2n_extension_type.h#L81-L84) to tell the difference and use the appropriate search strategy.